nixos/modules/virtualisation/digital-ocean-image.nix

```diff
@@ -0,0 +1,65 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.virtualisation.digitalOceanImage;
+in
+{
+
+  imports = [ ./digital-ocean-config.nix ];
+
+  options = {
+    virtualisation.digitalOceanImage.diskSize = mkOption {
+      type = with types; int;
+      default = 4096;
+      description = ''
+        Size of disk image. Unit is MB.
+      '';
+    };
+
+    virtualisation.digitalOceanImage.configFile = mkOption {
+      type = with types; nullOr path;
+      default = null;
+      description = ''
+        A path to a configuration file which will be placed at `/etc/nixos/configuration.nix`
+        and be used when switching to a new configuration.
+        If set to `null`, a default configuration is used that imports
+        `(modulesPath + "/virtualisation/digital-ocean-config.nix")`.
+      '';
+    };
+
+    virtualisation.digitalOceanImage.compressionMethod = mkOption {
+      type = types.enum [ "gzip" "bzip2" ];
+      default = "gzip";
+      description = ''
+        Disk image compression method.
```

Comment:
What is this option useful for?

Reply:
These are just the two compression formats that DO supports (unfortunately nothing better than bz2 is available). You're charged monthly for the size of the images stored in your DO account, so the tradeoff of using slower compression saves money as well as upload time.

Reply:
I see, maybe you could mention this in the description, along with which format is better for what